test/e2e/devices/deviceupdatetest/device_update_test.go

```diff
@@ -42,14 +42,31 @@ func TestCli_Devices_Update(t *testing.T) {
 					t.Fatal(err)
 				}
 				if status == true {
-					root.SetArgs([]string{subCommand, "update", "-i", device.GetId(), "-H", "metal-cli-update-dev-test", "-d", "This device used for testing"})
+					root.SetArgs([]string{subCommand, "update", "-i", device.GetId(), "-H", "metal-cli-update-dev-test", "-d", "This device used for testing", "--locked=true"})
 
 					out := helper.ExecuteAndCaptureOutput(t, root)
 
 					if !strings.Contains(string(out[:]), "metal-cli-update-dev-test") {
 						t.Error("expected output should include metal-cli-update-dev-test in the out string ")
 					}
 				}
+
+				root.SetArgs([]string{subCommand, "delete", "-i", device.GetId()})
+
+				out := helper.ExecuteAndCaptureOutput(t, root)
+
+				if !strings.Contains(string(out[:]), "not delete") {
+					t.Error("expected output should include 'not delete' in the out string due to device locking")
```

Comment:
This test is failing here; haven't checked this PR out yet to get a better idea of which part of this is failing (Device is not locked?  Device is locked but the error message doesn't match the check on line 58?  Something else?).

https://github.com/equinix/metal-cli/actions/runs/10741510710/job/29792085636?pr=485#step:5:77

Reply:
The delete subcommand waits for confirmation before deleting: https://github.com/equinix/metal-cli/blob/main/internal/devices/delete.go#L40-L41

This test will need to delete with the `-f` flag to bypass the confirmation prompt.